src/telliot_core/contract/contract.py

```diff
@@ -107,15 +109,62 @@ async def write(
             contract_function = self.contract.get_function_by_name(func_name)
             transaction = contract_function(**kwargs)
 
-            built_tx = transaction.buildTransaction(
-                {
-                    "from": acc.address,
-                    "nonce": acc_nonce,
-                    "gas": gas_limit,
-                    "gasPrice": self.node.web3.toWei(gas_price, "gwei"),
-                    "chainId": self.node.chain_id,
-                }
-            )
+            # start tx dict with static elements
+            tx_dict = {
+                "from": acc.address,
+                "nonce": acc_nonce,
+                "gas": gas_limit,
+                "chainId": self.node.chain_id,
+            }
+
+            # use legacy gas strategy if only legacy gas price is provided
```

Comment:
Maybe change the comment to something about using legacy transaction type, as the comment is fairly redundant given the readability of `if legacy_gas_price is not None`.